### ISSUE TITLE
Screen Reader: Many Pages: The buttons of date picker are announced without visible labels

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77559]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS or TalkBack on Android, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Date of birth menu item and double-tap to activate it
5. When the Date of birth picker opens, note how the year picker button and the month text are announced

Other Occurrences:
1. Create Expense > Confirm details > Date: same issue
2. Reports > Filters > Date > On/After/Before: same issue

## Expected Result:
The year button and month text should be announced using the same text that is visually displayed, e.g.:
- '2024, select year'
- 'November, select month'

## Actual Result:
The buttons of date picker are announced without visible labels.

## Area issue was found in:
Profile Subsections

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Update the accessibilityLabel to include the visible text, e.g.:
- '2024, select year'
- 'November, select month'
https://appt.org/en/docs/react-native/samples/accessibility-label
https://reactnative.dev/docs/accessibility#accessibilitylabel

## Environment:
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

![Image](https://github.com/user-attachments/assets/978793ee-36e3-4de7-a4c7-4de736a94f47)

https://github.com/user-attachments/assets/36a5ee3f-0e64-4036-88e9-7e7cba7ee3ee

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005782
**Issue reported by:** NA
**Slack conversation:** NA
